crates/biome_js_analyze/src/semantic_analyzers/correctness/use_exhaustive_dependencies.rs

```diff
@@ -519,6 +529,32 @@ fn function_of_hook_call(call: &JsCallExpression) -> Option<JsSyntaxNode> {
     })
 }
 
+// check if identifier is declared outside of function scope
+fn is_out_of_function_scope(
+    identifier: &AnyJsExpression,
+    function: &JsSyntaxNode,
+    model: &SemanticModel,
+) -> bool {
```

Comment:
In many functions inside the analyzer, we return `Option<bool>`. By changing this type, can simplify the code of the rule by a lot, for example:

```rs
let identifier_name = JsIdentifierExpression::cast_ref(identifier.syntax())?.name().ok()?;
```